File: MetodeAvansate/Siteuri/blog-app/BlogApp/BlogApp/Services/ArticlesService.cs
using BlogApp.Models;
using BlogApp.Repositories;

namespace BlogApp.Services
{
    public class ArticlesService : IArticlesService
    {
        private readonly IArticlesRepository _articlesRepository;

        public ArticlesService(IArticlesRepository articlesRepository)
        {
            _articlesRepository = articlesRepository;
        }

        public List<Article> GetAllArticles()
        {
            return _articlesRepository.GetAllArticles();
        }

        public Article GetArticle(int id)
        {
            return _articlesRepository.GetArticle(id);
        }

        public Article PostArticle(Article article)
        {
            return _articlesRepository.PostArticle(article);
        }

        public void EditArticle(int id, Article article)
        {
            _articlesRepository.EditArticle(id, article);
        }

        public Article DeleteArticle(int id)
        {
            return _articlesRepository.DeleteArticle(id);
        }
    }
}

File: MetodeAvansate/Siteuri/blog-app/blog-app/src/app/services/articles.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Article } from '../models/article';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ArticlesService {

  baseURL: string = "https://localhost:7107/";
  apiPath: string = "api/articles/";

  constructor(private httpClient: HttpClient) { }

  getArticles(search: string): Observable<Article[]> {
    return this.httpClient.get<Article[]>(`${this.baseURL}${this.apiPath}?search=${search}`);
  }

  getOneArticle(id: number): Observable<Article> {
    return this.httpClient.get<Article>(`${this.baseURL}${this.apiPath}${id}`);
  }

  postArticle(article: Article): Observable<Article> {
    return this.httpClient.post<Article>(`${this.baseURL}${this.apiPath}`, article);
  }

  editArticle(id: number, article: Article): Observable<any> {
    return this.httpClient.put<any>(`${this.baseURL}${this.apiPath}${id}`, article);
  }

  deleteArticle(id: number): Observable<Article> {
    return this.httpClient.delete<Article>(`${this.baseURL}${this.apiPath}${id}`);
  }
}
